### PR TITLE
Issue #2174 Check that InternetGateway exists before returning from creation

### LINKED ISSUE #2174: provider/aws: IGW sometimes takes time to create
When I try to bootstrap a non-trivial infrastructure with an IGW, I sometimes run into the following error:

```
module.yada_yada.aws_internet_gateway.default: Error: 1 error(s) occurred:

* InvalidInternetGatewayID.NotFound: The internetGateway ID 'igw-c30b7ba3' does not exist
	status code: 400, request id: []
```

I noticed that for some resources we check if these are created after we send the request for creating before doing anything else, so I guess similar approach may need to be applied here.

// btw. the same infrastructure works well on second run of `terraform apply`

### PR DESCRIPTION
**Pull request by carlossg**
Fix #2174 some random InvalidInternetGatewayID.NotFound errors, seems to happen during attachment, right after creation

The errors look like

```
21:28:49 Initializing environment terraform
21:28:52 [vpc] aws_vpc.myvpc: Creating...
21:28:52 [vpc]   cidr_block:                 "" => "10.16.0.0/16"
21:28:52 [vpc]   default_network_acl_id:     "" => "<computed>"
21:28:52 [vpc]   default_security_group_id:  "" => "<computed>"
21:28:52 [vpc]   dhcp_options_id:            "" => "<computed>"
21:28:52 [vpc]   enable_classiclink:         "" => "<computed>"
21:28:52 [vpc]   enable_dns_hostnames:       "" => "1"
21:28:52 [vpc]   enable_dns_support:         "" => "1"
21:28:52 [vpc]   main_route_table_id:        "" => "<computed>"
21:28:52 [vpc]   tags.#:                     "" => "3"
21:28:52 [vpc]   tags.Name:                  "" => "myvpc-testing"
21:28:52 [vpc] aws_vpc.myvpc: Creation complete
21:28:52 [vpc] aws_internet_gateway.myvpc: Creating...
21:28:52 [vpc]   tags.#:                     "0" => "3"
21:28:52 [vpc]   tags.Name:                  "" => "myvpc-testing"
21:28:52 [vpc]   vpc_id:                     "" => "vpc-c7e1f3a3"
21:28:52 [vpc] aws_subnet.myvpc: Creating...
21:28:52 [vpc]   availability_zone:          "" => "<computed>"
21:28:52 [vpc]   cidr_block:                 "" => "10.16.0.0/16"
21:28:52 [vpc]   map_public_ip_on_launch:    "" => "1"
21:28:52 [vpc]   tags.#:                     "" => "3"
21:28:52 [vpc]   tags.Name:                  "" => "myvpc-testing"
21:28:52 [vpc]   vpc_id:                     "" => "vpc-c7e1f3a3"
21:28:52 [vpc] aws_subnet.myvpc: Creation complete
21:28:52 [vpc] Error applying plan:
21:28:52 [vpc] 
21:28:52 [vpc] 1 error(s) occurred:
21:28:52 [vpc] 
21:28:52 
[vpc] * aws_internet_gateway.myvpc: InvalidInternetGatewayID.NotFound: The internetGateway ID 'igw-748d0310' does not exist
21:28:52 [vpc]  status code: 400, request id: 
```

### CODE REVIEW

File: builtin/providers/aws/resource_aws_internet_gateway.go [hunk 1]
```diff
@@ -50,6 +50,27 @@ func resourceAwsInternetGatewayCreate(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
+	resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DescribeInternetGateways(&ec2.DescribeInternetGatewaysInput{
```
Comment (catsby) [inline builtin/providers/aws/resource_aws_internet_gateway.go]: The response here should be able to tell us something useful about the newly created Internet Gateway, I'm not sure we should ignore it

**Comment by catsby** [comment]
Hey @carlossg thanks for the PR. Do you think we could re-use the `IGStateRefreshFunc` method for this?

File: builtin/providers/aws/resource_aws_internet_gateway.go (hunk 1 above)
Comment (carlossg) [inline builtin/providers/aws/resource_aws_internet_gateway.go]: just copied it from https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_api_gateway_resource.go#L132
Not sure it is useful, other than waiting for consistency in AWS

**Comment by carlossg** [comment]
@catsby something like that ?

**Comment by catsby** [comment]
LGTM, and the acceptance tests pass, thanks!

**Pull request merged**